source/validate_capability.cpp

```diff
@@ -77,6 +77,54 @@ bool IsSupportOptionalVulkan_1_0(uint32_t capability) {
   return false;
 }
 
+bool IsSupportGuaranteedOpenCL_1_2(uint32_t capability) {
+  switch (capability) {
+    case SpvCapabilityAddresses:
+    case SpvCapabilityFloat16Buffer:
+    case SpvCapabilityGroups:
+    case SpvCapabilityInt64:
```

Comment:
The embedded profile requires cles_khr_int64 in order to support Int64

Reply:
I'm ok with this if we interpret an OpenCL environment without qualification as the full profile.   That should be documented in the enum definition.

Reply:
I completely ignored the embedded profile for now, until we decided out to differentiate it from the full profile (if we wanted to make that distinction). I’ll update the series to handle the embedded profile as well.